lua/packer/luarocks.lua

```diff
@@ -0,0 +1,305 @@
+-- Add support for installing and cleaning Luarocks dependencies
+-- Based off of plenary/neorocks/init.lua in https://github.com/nvim-lua/plenary.nvim
+local a = require('packer.async')
+local jobs = require('packer.jobs')
+local log = require('packer.log')
+local result = require('packer.result')
+local util = require('packer.util')
+
+local fmt = string.format
+local async = a.sync
+local await = a.wait
+
+local lua_version = nil
+if jit then
+  local jit_version = string.gsub(jit.version, 'LuaJIT ', '')
+  lua_version = {lua = string.gsub(_VERSION, 'Lua ', ''), jit = jit_version, dir = jit_version}
+end
+
+local cache_path = vim.fn.stdpath('cache')
+local rocks_path = util.join_paths(cache_path, 'packer_hererocks')
+local hererocks_file = util.join_paths(rocks_path, 'hererocks.py')
+local hererocks_install_dir = util.join_paths(rocks_path, lua_version.dir)
+local _hererocks_setup_done = false
+local function hererocks_is_setup()
+  if _hererocks_setup_done then return true end
+  await(a.main)
+  _hererocks_setup_done = vim.fn.isdirectory(util.join_paths(hererocks_install_dir, 'lib')) > 0
```

Comment:
May want to use vim.loop.fs_stat here. In general I'd say you want to avoid as much `vim.fn` here in case people call it during weird situations. But probably works for now though :)

Reply:
Fair enough, the less overhead the better.